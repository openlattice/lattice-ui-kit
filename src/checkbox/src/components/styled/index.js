// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES } from '../../../../colors';

const CheckboxLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;


type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    position: absolute;
    left: 8px;
    top: 4px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  `;
};

const CheckboxIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 3px;

  :after {
    ${getAfterStyles}
    }
  }
`;

const CheckboxInput = styled.input.attrs(() => ({
  type: 'checkbox'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
  
  :hover + ${CheckboxIndicator},
  :focus + ${CheckboxIndicator} {
    background: ${NEUTRALS[1]};
  }

  :checked + ${CheckboxIndicator} {
    background: ${PURPLES[2]};
  }

  :checked:hover + ${CheckboxIndicator},
  :checked:focus + ${CheckboxIndicator} {
    background: ${PURPLES[1]};
  }

  :disabled + ${CheckboxIndicator} {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + ${CheckboxIndicator} {
    background: ${NEUTRALS[2]};
  }
`;


export {
  CheckboxIndicator,
  CheckboxInput,
  CheckboxLabel,
};
