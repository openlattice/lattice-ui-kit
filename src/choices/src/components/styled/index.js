// @ flow
import CheckboxIndicator from './CheckboxIndicator';
import ChoiceLabel from './ChoiceLabel';
import RadioIndicator from './RadioIndicator';
import { CheckboxInput, RadioInput } from './ChoiceInputs';

export {
  CheckboxIndicator,
  CheckboxInput,
  ChoiceLabel,
  RadioIndicator,
  RadioInput
};
