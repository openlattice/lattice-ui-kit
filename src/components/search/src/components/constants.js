import { List, Map, fromJS } from 'immutable';

const mockFilterFields = [
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'reportType',
    label: 'Report type',
    options: ['Crisis Template', 'Follow-up'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'badges',
    label: 'Badges',
    options: ['Officer Safety', 'Substance use'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'submitter',
    label: 'Submitter',
    options: [
      'squidward@bubblebowl.com',
      'smitty@werbenjagermanjensen.com',
      'wumbology@thestudyofwumbo.com',
      'enigma@innermachinations.com'
    ],
  },
];

const mockSearchResultsForReports = List([
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety'],
    submitter: 'squidward@bubblebowl.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'squidward@bubblebowl.com',
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'smitty@werbenjagermanjensen.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Substance use'],
    submitter: 'wumbology@thestudyofwumbo.com'
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'enigma@innermachinations.com'
  }),
  Map({
    reportType: 'Follow-up',
    submitter: 'enigma@innermachinations.com'
  }),
]);

const mockSearchResultsForPeople = List([
  Map({
    lastName: 'Werbenjagermanjensen',
    firstName: 'Smitty',
    middleName: '',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Fish',
    dob: '02/22/2002',
    identifier: '365ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Thornton',
    firstName: 'Cameron',
    middleName: 'Orville',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Black / African American',
    dob: '01/05/1982',
    identifier: '565ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Gerner',
    firstName: 'Timothy',
    middleName: 'Pernell',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Native Hawaiian or Other Pacific Islander',
    dob: '10/02/1991',
    identifier: '465ba027-0c71-47d1-a5f4-61346ed96d8c',
  })
]);

export {
  mockFilterFields,
  mockSearchResultsForReports,
  mockSearchResultsForPeople
};
