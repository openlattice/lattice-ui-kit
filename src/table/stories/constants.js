const TABLE_HEADERS = [
  { key: 'name', label: 'Subject Name', cellStyle: { width: '50%' } },
  { key: 'dob', label: 'Date of Birth' },
  { key: 'manager', label: 'Manager' },
  { key: 'lastUpdated', label: 'Last Updated' },
  { key: 'id', label: 'ID', sortable: false },
];

const TABLE_DATA = [
  {
    name: 'Eugene Krabs',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-09-01',
    id: '1'
  },
  {
    name: 'Squidward Tentacles',
    dob: '',
    manager: 'Smitty',
    lastUpdated: '2019-08-29',
    id: '2'
  },
  {
    name: 'Sandy Cheeks',
    dob: '1987-11-17',
    manager: 'Patrick',
    lastUpdated: '2019-07-30',
    id: '3'
  },
  {
    name: 'Larry the Lobster',
    dob: '1975-06-23',
    manager: 'Patchy',
    lastUpdated: '2019-08-16',
    id: '4'
  },
  {
    name: 'Sheldon Plankton',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-08-20',
    id: '5'
  },
  {
    name: 'Mrs. Puff',
    dob: '',
    manager: 'Spongebob',
    lastUpdated: '2019-08-21',
    id: '6'
  },
  {
    name: 'Flying Dutchman',
    dob: '1678-04-20',
    manager: 'Spongebob',
    lastUpdated: '2019-06-21',
    id: '7'
  },
];

export {
  TABLE_HEADERS,
  TABLE_DATA,
};
