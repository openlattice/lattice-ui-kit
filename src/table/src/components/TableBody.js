// @flow
import React from 'react';

import Spinner from '../../../spinner';
import { getSortedData } from './TableUtils';

import { StyledRow, Cell } from './styled';
import type { RowData, SortOrder } from '../../types';

type Props = {
  className ? :string;
  components :Object;
  data :RowData[];
  exact ? :boolean;
  headers :Object[];
  isLoading ? :boolean;
  order ? :SortOrder;
  orderBy ? :string;
  page :number;
  rowsPerPage :number;
};

const TableBody = (props :Props) => {
  const {
    className,
    components,
    data,
    headers,
    isLoading,
    exact,
    order,
    orderBy,
    page,
    rowsPerPage,
  } = props;

  const sortedData = getSortedData(headers, data, order, orderBy);
  const dataByPage :RowData[] = exact
    ? sortedData
    : sortedData.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage);

  // inject empty row to maintain table size
  const emptyRowCount = rowsPerPage - dataByPage.length;
  // height per row + (row - 1) * (padding + border)
  let emptyHeight = `${emptyRowCount * 24 + (emptyRowCount - 1) * 21}px`;

  if (isLoading) {
    emptyHeight = `${rowsPerPage * 24 + (rowsPerPage - 1) * 21}px`;
    return (
      <tbody className={className}>
        <StyledRow id="empty-row-filler">
          <Cell colSpan={headers.length} cellStyle={{ height: emptyHeight, textAlign: 'center' }}>
            {
              isLoading && (
                <Spinner />
              )
            }
          </Cell>
        </StyledRow>
      </tbody>
    );
  }

  return (
    <tbody className={className}>
      {
        dataByPage.map((rowData :RowData) => {

          const { id } = rowData;
          return (
            <components.Row
                key={id}
                data={rowData}
                headers={headers}
                components={components} />
          );
        })
      }
      {
        !!emptyRowCount && (
          <StyledRow id="empty-row-filler">
            <Cell colSpan={headers.length} cellStyle={{ height: emptyHeight }} />
          </StyledRow>
        )
      }
    </tbody>
  );
};

TableBody.defaultProps = {
  className: undefined,
  data: [],
  isLoading: false,
  exact: false,
  order: false,
  orderBy: undefined
};

export default React.memo<Props>(TableBody);
