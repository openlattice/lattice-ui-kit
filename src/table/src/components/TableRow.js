// @flow

import { StyledRow } from './styled';
import type { RowData } from '../../types';

type Props = {
  className ? :string;
  components :Object;
  data :RowData;
  headers :Object[];
};

const TableRow = (props :Props) => {
  const {
    className,
    components,
    data,
    headers
  } = props;

  const { id } = data;

  const cells = headers
    .map((header) => <components.Cell key={`${id}_cell_${header.key}`}>{data[header.key]}</components.Cell>);

  return (
    <StyledRow className={className}>
      {cells}
    </StyledRow>
  );
};

TableRow.defaultProps = {
  className: undefined
};

export default TableRow;
