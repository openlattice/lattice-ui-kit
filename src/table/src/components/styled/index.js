// @flow
import styled from 'styled-components';

import { NEUTRALS, WHITE } from '../../../../colors';
import { getStickyPosition } from '../../../../utils/StyleUtils';

const StyledTable = styled.table`
  background-color: ${WHITE};
  border-collapse: collapse;
  border: none;
  table-layout: fixed;
  width: 100%;
`;

const PaginationWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: flex-end;
  width: 100%;
`;

const Cell = styled.td((props :Object) => ({
  backgroundColor: props.as === 'th' ? NEUTRALS[6] : 'inherit',
  cursor: props.onClick ? 'pointer' : 'auto',
  padding: '10px 10px',
  textAlign: 'left',
  wordWrap: 'break-word',
  ...props.cellStyle
}));

Cell.displayName = 'Cell';

const StyledRow = styled.tr`
  background-color: ${WHITE};
  border-bottom: 1px solid ${NEUTRALS[4]};

  /* position: sticky doesn't work with thead/tr. Move to all cells within */
  td, th {
    ${getStickyPosition}
  };
`;

const RowPerPageWrapper = styled.div`
  font-size: 14px;
  margin-right: 10px;
  width: 75px;
`;

export {
  Cell,
  PaginationWrapper,
  RowPerPageWrapper,
  StyledRow,
  StyledTable,
};
