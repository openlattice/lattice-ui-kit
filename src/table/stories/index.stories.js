import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';

import { Card, CardSegment } from '../../layout';
import { TABLE_DATA, TABLE_HEADERS } from './constants';
import { getHoverStyles } from '../../utils/StyleUtils';
import {
  Table
} from '..';

const CustomRow = styled.tr`
  ${getHoverStyles};
`;

const components = {
  Row: () => (
    <CustomRow onClick={() => {}}>
      <td colSpan="4">row</td>
    </CustomRow>
  )
};

storiesOf('Table', module)
  .add('Sortable', () => (
    <Card>
      <CardSegment vertical>
        <Table
            headers={TABLE_HEADERS}
            data={TABLE_DATA} />
      </CardSegment>
    </Card>
  ))
  .add('Paginated', () => (
    <Card>
      <CardSegment vertical>
        <Table
            headers={TABLE_HEADERS}
            data={TABLE_DATA}
            rowsPerPageOptions={[5, 20, 50]}
            paginated />
      </CardSegment>
    </Card>
  ))
  .add('Custom Components', () => (
    <Card>
      <CardSegment vertical>
        <Table
            components={components}
            headers={TABLE_HEADERS}
            data={TABLE_DATA}
            rowsPerPageOptions={[5, 20, 50]}
            paginated />
      </CardSegment>
    </Card>
  ));
