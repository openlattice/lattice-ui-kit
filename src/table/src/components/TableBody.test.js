import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TableBody from './TableBody';
import TableRow from './TableRow';
import { Cell } from './styled';
import { TABLE_DATA, TABLE_HEADERS } from '../../stories/constants';

describe('TableBody', () => {

  const components = { Row: TableRow, Cell };

  describe('render', () => {
    test('should render tbody at root', () => {
      const wrapper = shallow(
        <TableBody
            components={components}
            headers={TABLE_HEADERS}
            data={TABLE_DATA}
            rowsPerPage={10}
            page={0} />
      );
      expect(wrapper.type()).toEqual('tbody');
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should render spinner when isLoading=true', () => {
      const wrapper = shallow(
        <TableBody
            components={components}
            headers={TABLE_HEADERS}
            isLoading
            data={[]}
            rowsPerPage={10}
            page={0} />
      );

      expect(wrapper.find('Spinner')).toHaveLength(1);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should render sorted data by page', () => {
      const rowComponent = () => <div>test</div>;
      const wrapper = shallow(
        <TableBody
            components={components}
            data={TABLE_DATA}
            headers={TABLE_HEADERS}
            order="desc"
            orderBy="name"
            rowsPerPage={5}
            rowComponent={rowComponent}
            page={0} />
      );

      let rowComponents = wrapper.find(rowComponent);
      let actualOrderByIndices = [1, 4, 2, 5, 3];
      rowComponents.forEach((row, index) => {
        const rowData = row.prop('data');
        const actualIndex = actualOrderByIndices[index];
        expect(rowData).toEqual(TABLE_DATA[actualIndex]);
      });

      // change to page: 1
      wrapper.setProps({ page: 1 });
      rowComponents = wrapper.find(rowComponent);
      actualOrderByIndices = [6, 0];
      rowComponents.forEach((row, index) => {
        const rowData = row.prop('data');
        const actualIndex = actualOrderByIndices[index];
        expect(rowData).toEqual(TABLE_DATA[actualIndex]);
      });
    });

    test('should render empty-row-filler if displayed rows is less then rowsPerPage', () => {
      const wrapper = shallow(
        <TableBody
            components={components}
            data={TABLE_DATA}
            headers={TABLE_HEADERS}
            rowsPerPage={5}
            page={0} />
      );

      expect(wrapper.find('#empty-row-filler')).toHaveLength(0);
      wrapper.setProps({ page: 1 });
      expect(wrapper.find('#empty-row-filler')).toHaveLength(1);
    });
  });
});
