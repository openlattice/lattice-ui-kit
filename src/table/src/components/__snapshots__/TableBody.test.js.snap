// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody render should render spinner when isLoading=true 1`] = `
<tbody>
  <styled__StyledRow
    id="empty-row-filler"
  >
    <Cell
      cellStyle={
        Object {
          "height": "429px",
          "textAlign": "center",
        }
      }
      colSpan={5}
    >
      <Spinner
        bottomColor="#cdd1db"
        centered={true}
        size="2x"
        topColor="#555e6f"
      />
    </Cell>
  </styled__StyledRow>
</tbody>
`;

exports[`TableBody render should render tbody at root 1`] = `
<tbody>
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "1942-11-30",
        "id": "1",
        "lastUpdated": "2019-09-01",
        "manager": "Spongebob",
        "name": "Eugene Krabs",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row0-id1"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "",
        "id": "2",
        "lastUpdated": "2019-08-29",
        "manager": "Smitty",
        "name": "Squidward Tentacles",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row1-id2"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "1987-11-17",
        "id": "3",
        "lastUpdated": "2019-07-30",
        "manager": "Patrick",
        "name": "Sandy Cheeks",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row2-id3"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "1975-06-23",
        "id": "4",
        "lastUpdated": "2019-08-16",
        "manager": "Patchy",
        "name": "Larry the Lobster",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row3-id4"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "1942-11-30",
        "id": "5",
        "lastUpdated": "2019-08-20",
        "manager": "Spongebob",
        "name": "Sheldon Plankton",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row4-id5"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "",
        "id": "6",
        "lastUpdated": "2019-08-21",
        "manager": "Spongebob",
        "name": "Mrs. Puff",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row5-id6"
  />
  <TableRow
    components={
      Object {
        "Cell": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__Cell-r3xwjs-2",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Cell",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__Cell-r3xwjs-2",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Row": [Function],
      }
    }
    data={
      Object {
        "dob": "1678-04-20",
        "id": "7",
        "lastUpdated": "2019-06-21",
        "manager": "Spongebob",
        "name": "Flying Dutchman",
      }
    }
    headers={
      Array [
        Object {
          "cellStyle": Object {
            "width": "50%",
          },
          "key": "name",
          "label": "Subject Name",
        },
        Object {
          "key": "dob",
          "label": "Date of Birth",
        },
        Object {
          "key": "manager",
          "label": "Manager",
        },
        Object {
          "key": "lastUpdated",
          "label": "Last Updated",
        },
        Object {
          "key": "id",
          "label": "ID",
          "sortable": false,
        },
      ]
    }
    key="row6-id7"
  />
  <styled__StyledRow
    id="empty-row-filler"
  >
    <Cell
      cellStyle={
        Object {
          "height": "114px",
        }
      }
      colSpan={5}
    />
  </styled__StyledRow>
</tbody>
`;
